### PR TITLE
Add support for PodSets and other improvements

### PR DESCRIPTION
**Pull request by fvaleri**
Fixes and improvements related to #6476, #6477, #6471.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Can you please rebase it? It seems ot have the issue with the `tar`alias.

**Comment by scholzj** [comment]
It looks like after the backup, the collected files are tared, but the temporary directory structure and its data remains (in my case `myproject/my-cluster` - `namespace/cluster-name`). Is that intentional? I guess it should be deleted, or?

**Comment by fvaleri** [comment]
> It looks like after the backup, the collected files are tared, but the temporary directory structure and its data remains (in my case `myproject/my-cluster` - `namespace/cluster-name`). Is that intentional? I guess it should be deleted, or?

It is like this from the start. There is no `rm` command in the entire script. I would prefer the user to explicitly delete that staging folder before running a new backup. Initially, I also thought about supporting incremental backups with `rsync`, but this does not work well in practice. A possible improvements my be to use a more advanced tool such as `rclone`, but this is outside the scope of this PR.

**Comment by scholzj** [comment]
> It is like this from the start. There is no rm command in the entire script. I would prefer the user to explicitly delete that staging folder before running a new backup.

Ok, I din't remembered it from before, so I was wondering if it is some bug. But if it is intentional its fine.

**Comment by scholzj** [comment]
Some more comments ... I observed them when testing with StrimziPodSets -> not sure if they apply to StatefulSets
* It does not wait for the pods to be terminated. As a result, it tries to delete the Kafka pods first, but because if does not wait, it deletes Zoo right after. That Zoo shutdowns faster than Kafka and leaves the Kafka pod in unclean shutdown. which is bad and should be avoided (I can open a new issue for it if you would want to deal with it separately).
* The cluster does not seem to recover after the backup. It tries to first recover the EO. But that will not work before the Zoo/Kafka pods are ready. And they will not be ready unless you enable the CO first. But without them, the EO never gets ready and the whole backup fails with this:
   ```
   error: timed out waiting for the condition on pods/my-cluster-entity-operator-8554c4bf6c-rxtn9
   backup failed at 144: $con_cmd
   ```
   So I think the right approach here would be to first enable the CO and let the CO recover the cluster. That should work both  with StatefulSets as well as without StatefulSets.
* On one occasion, I got following error:
   ```
    ...
    ./kafka-log2/__strimzi-topic-operator-kstreams-topic-store-changelog-0/00000000000000000000.index
    ./kafka-log2/__strimzi-topic-operator-kstreams-topic-store-changelog-0/00000000000000000000.timeindex
    gtar: Unexpected EOF in archive
    gtar: Unexpected EOF in archive
    gtar: Error is not recoverable: exiting now
    pod "cold-backup" deleted
    Compressing xxx/myproject/my-cluster to xxx/xxx.tgz
    ...
   ```
   Not sure what caused this -> raising it just in case you saw it in the past.

The changes related to the other 2 issues seem to work fine here.

**Comment by fvaleri** [comment]
I agree on the first two points, and I'll commit required changes in this PR. I haven't seen any tar error since first experiments. Feel free to open a separate issue in case you can reproduce it consistently.

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
> I agree on the first two points, and I'll commit the required changes in this PR. I haven't seen any tar error since first experiments. Feel free to open a separate issue in case you can reproduce it consistently.

I saw it just once so far. Maybe it was some network glitch between my server and local machine running the tool or something.

**Comment by scholzj** [comment]
@fvaleri I forgot about it yesterday -> but since it should now work with StrimziPodSets, can you please also remove the `@StatefulSetTest` annotation from the System Tests? So that I can run them?

**Comment by fvaleri** [comment]
Sure, and thanks for taking the time to review this.

File: tools/cold-backup/README.md [hunk 1]
```diff
@@ -49,15 +55,15 @@ kubectl run producer-perf -it --image="$CLIENT_IMAGE" --rm="true" --restart="Nev
     --throughput -1 --producer-props acks=1 bootstrap.servers="my-cluster-kafka-bootstrap:9092"
 
 # run backup procedure
-./tools/cold-backup/run.sh backup -n myproject -c my-cluster -t /tmp/my-cluster.zip
+./tools/cold-backup/run.sh backup -n myproject -c my-cluster -t /tmp/my-cluster.tgz
```
Comment (ppatierno) [inline tools/cold-backup/README.md:58]: by changing to tgz I would say that requirements in the README could be changed by removing the zip and unzip as needed?

**Review by ppatierno — APPROVED** [review]
Just one nit but LGTM.

File: tools/cold-backup/README.md (hunk 1 above)
Comment (fvaleri) [reply tools/cold-backup/README.md:58]: Right, thanks for spotting this.

**Comment by scholzj** [comment]
Thanks for the PR @fvaleri

**Pull request merged**